File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    a1 = [1,2,3,4,5] ;i1 = 0
    b1 = [2,1,2,3,2,4,2,5]; i2 = 0
    c1 = [3,3,1,1,2,2,4,4,5,5]; i3 = 0
    
    tmp = [0, 0, 0]
    
    for ans in answers:
        if ans == a1[i1]: tmp[0] += 1
        if ans == b1[i2]: tmp[1] += 1
        if ans == c1[i3]: tmp[2] += 1
        i1 = (i1+1) % 5
        i2 = (i2+1) % 8
        i3 = (i3+1) % 10
        
    max_t = max(tmp); res = []
    for i in range(3):
        if tmp[i] == max_t:
            res.append(i+1)
    res.sort()
    return res

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> v1 = {1,2,3,4,5}; int a1=0;
    vector<int> v2 = {2,1,2,3,2,4,2,5}; int a2=0;
    vector<int> v3 = {3,3,1,1,2,2,4,4,5,5}; int a3=0;
    
    for(int i = 0; i < answers.size(); i++){
        if (answers[i] == v1[i%v1.size()]) a1++;
        if (answers[i] == v2[i%v2.size()]) a2++;
        if (answers[i] == v3[i%v3.size()]) a3++;
    }
    int max_score = max({a1,a2,a3});
    if(a1==max_score) answer.push_back(1);
    if(a2==max_score) answer.push_back(2);
    if(a3==max_score) answer.push_back(3);
    
    return answer;
}